File: DSA/VedantChouhan_590024563/Day-03/q2.c
#include <stdio.h>

int firstOccurrence(int arr[], int n, int target) {
    int low = 0, high = n - 1;
    int ans = -1;

    while (low <= high) {
        int mid = low + (high - low) / 2;

        if (arr[mid] == target) {
            ans = mid;
            high = mid - 1; // search left side
        }
        else if (arr[mid] < target) {
            low = mid + 1;
        }
        else {
            high = mid - 1;
        }
    }

    return ans;
}

int lastOccurrence(int arr[], int n, int target) {
    int low = 0, high = n - 1;
    int ans = -1;

    while (low <= high) {
        int mid = low + (high - low) / 2;

        if (arr[mid] == target) {
            ans = mid;
            low = mid + 1; // search right side
        }
        else if (arr[mid] < target) {
            low = mid + 1;
        }
        else {
            high = mid - 1;
        }
    }

    return ans;
}

int main() {
    int arr[] = {1, 2, 2, 2, 3, 4, 5};
    int n = sizeof(arr) / sizeof(arr[0]);
    int target = 2;

    int first = firstOccurrence(arr, n, target);

    if (first == -1) {
        printf("Occurrences: 0\n");
        return 0;
    }

    int last = lastOccurrence(arr, n, target);

    printf("Occurrences: %d\n", last - first + 1);

    return 0;
}

File: DSA/vaishnavi_joshi_590016947/Day3/Q2.java
public class NumberOfOccurance {

    public static int countOccurrences(int[] arr, int target) {

        int count = 0;

        for (int i = 0; i < arr.length; i++) {
            if (arr[i] == target) {
                count++;
            }
        }

        return count;
    }

    public static void main(String[] args) {

        int[] arr = {1, 1, 2, 2, 2, 2, 4, 4};
        int target = 2;

        int result = countOccurrences(arr, target);

        System.out.println("Occurrences = " + result);
    }
}